src/content.js

```diff
@@ -37,45 +37,81 @@ async function getTagMetadata(tagsDir) {
   return defsByName;
 }
 
+function getPagePathParts(contentDir, readmeFilePath) {
+  const {dir} = path.parse(readmeFilePath);
+  const contentDirDepth = path.normalize(contentDir).split(path.sep).length;
+  return path.normalize(dir).split(path.sep).slice(contentDirDepth);
+}
+
+//return an array of the metadata objects representing each content page
 async function getPageMetadata(contentDir) {
-  const filePaths = await findPaths(path.join(contentDir, "**/readme.md"));
-  return await Promise.all(filePaths.map(async (filePath) => {
+  //find all readme files under the content root -- each will become a page
+  const readmeFilePaths = await findPaths(path.join(contentDir, "**/readme.md"));
+
+  //build the array of per-page metadata
+  const pages = await Promise.all(readmeFilePaths.map(async (readmeFilePath) => {
     return new Promise(async (resolve, reject) => {
       try {
-        const fileContents = await fs.readFile(filePath, "utf8");
-        const {dir} = path.parse(filePath);
+        const fileContents = await fs.readFile(readmeFilePath, "utf8");
         const {attributes, body} = fm(fileContents);
         if (!attributes || !attributes.title) {
-          reject(new Error(`File ${filePath} does not define a title`));
+          reject(new Error(`File ${readmeFilePath} does not define a title`));
         } else {
-          const contentDirDepth = path.normalize(contentDir).split(path.sep).length;
-          const _dir = path.normalize(dir).split(path.sep).slice(contentDirDepth);
+          const _pathParts = getPagePathParts(contentDir, readmeFilePath);
           const _headers = findHeaders(body);
           resolve({
+            //include any YAML metadata provided by page authors
             ...attributes,
+            //add calulated or parsed metadata
             _md: body,
             _headers,
-            _dir,
-            _slug: _dir[_dir.length - 1],
-            _dirUrl: "/" + _dir.join("/"),
+            _pathParts,
+            _slug: _pathParts[_pathParts.length - 1],
+            _path: "/" + _pathParts.join("/"),
           });
         }
       } catch (err) {
         reject(err);
       }
     });
   }));
+
+  //in a second pass, add inter-page relationships like parents and children
+  for (let page of pages) {
+    //find the parent of this page
+    for (let i = page._pathParts.length - 1; i >= 0; i--) {
+      const parentPath = "/" + page._pathParts.slice(0, i).join("/");
+      page._parent = pages.find(otherPage => otherPage._path == parentPath);
+      if (page._parent) {
+        break;
+      }
+    }
+
+    page._childPages = [];
+    const candidateChildPages = pages.filter(otherPage =>
+      otherPage._path != page._path && otherPage._path.startsWith(page._path)
+    );
+    candidateChildPages.sort((a, b) => a._pathParts.length - b._pathParts.length);
```

Comment:
Sorting candidate child pages by their path means shorter paths end up in the final results first, then transitive children of children can be easily ignored by checking if their path starts with a page already added to the results.